CONTRIBUTING.md

```diff
@@ -1,62 +1,21 @@
+in [ARCHITECTURE.md](https://github.com/kneasle/wheatley/blob/better-contributing/ARCHITECTURE.md).
 
-## Code structure
+## Testing Your Changes
+
+To run Wheatley **from source code**, `cd` to the repository root and run:
+```bash
+python run-wheatley [ARGS]
+# or possibly
+python3 run-wheatley [ARGS]
 ```
-.
-├── CHANGE_LOG.md
-├── CONTRIBUTING.md
-├── LICENSE
-├── README.md
-├── requirements.txt
-├── run-wheatley                                    => A shell-executable python script to launch Wheatley
-├── setup.py                                        => Build script run to generate the PIP package
-├── wheatley
-│   ├── __init__.py                                 => Empty file
-│   ├── __main__.py                                 => Code to start Wheatley
-│   ├── arg_parsing.py                              => Code to parse formatted CLI args e.g. call strings
-│   ├── bell.py                                     => Stores the representation of a bell
-│   ├── bot.py                                      => The main class for Wheatley
-│   ├── calls.py                                    => Constants for the calls used in Ringing Room
-│   ├── main.py                                     => Entry point of Wheatley
-│   ├── page_parser.py                              => Code to parse the ringing room HTML
-│   ├── regression.py                               => Code for the weighted linear regression
-│   ├── rhythm.py                                   => Code used to drive Wheatley's rhythm
-│   ├── tower.py                                    => Code used to interact with ringing room
-│   └── row_generation                              => Modular row generators
-│       ├── __init__.py
-│       ├── complib_composition_generator.py
-│       ├── dixonoids_generator.py
-│       ├── go_and_stop_calling_generator.py
-│       ├── helpers.py
-│       ├── method_place_notation_generator.py
-│       ├── place_notation_generator.py
-│       ├── plain_hunt_generator.py
-│       └── row_generator.py
-├── tests
-│   ├── __init__.py
-│   ├── test_arg_parsing.py                         => Unit tests for CLI arg parsing
-│   └── row_generation                              => Unit tests for the row generation module
-│       ├── __init__.py
-│       ├── generator_test_helpers.py
-│       ├── test_DixonoidsGenerator.py
-│       ├── test_Helpers.py
-│       ├── test_MethodPlaceNotationGenerator.py
-│       ├── test_PlaceNotationGenerator.py
-│       └── test_PlainHuntGenerator.py
-├── fuzz                                            => Shell-executable script to run all the fuzzing
-└── fuzzing
-    ├── __init__.py
-    ├── call_parsing.py
-    └── fuzz_utils.py
- ```
+Or, on Unix you can run `./run-wheatley [ARGS]`.  Basically, replacing `wheatley` with
+`python run-wheatley` or `./run-wheatley` will have exactly the same effect but will use your
+updated source code rather than the version installed by `pip`.
```

Comment:
Looks really good.

I did add a section in around what linting there is and how to run it.
`## Linting and testing
The following should be run before committing to ensure code integrity:

Linter:
```
py -m pylint wheatley
```

Typechecker:
```
py -m mypy wheatley --pretty --disallow-incomplete-defs --disallow-untyped-defs
```

Unit tests
```
py -m pytest
```
`
Not sure if you think that might be worth keeping too as I don't think it's covered from my read through

Reply:
Yep I've added it now